test/src/state_machine/state.test.cpp

```diff
@@ -0,0 +1,781 @@
+/*
+ * Author: Efe Ozbatur, Franz Miltz
+ * Organisation: HYPED
+ * Date: 06.04.2021
+ *
+ *    Copyright 2019 HYPED
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ *    except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *    either express or implied. See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <fcntl.h>
+#include <stdlib.h>
+
+#include <random>
+#include <string>
+#include <vector>
+
+#include "data/data.hpp"
+#include "gtest/gtest.h"
+#include "state_machine/state.hpp"
+#include "state_machine/transitions.hpp"
+#include "utils/logger.hpp"
+
+using namespace hyped::data;
+using namespace hyped::state_machine;
+
+struct StateTest : public ::testing::Test {
+  // ---- Logger ---------------
+
+  hyped::utils::Logger log;
+  int stdout_f;
+  int tmp_stdout_f;
+
+  // ---- Test size -----------
+
+  static constexpr int TEST_SIZE = 1000;
+
+ protected:
+  void SetUp()
+  {
+    fflush(stdout);
+    stdout_f     = dup(1);
+    tmp_stdout_f = open("/dev/null", O_WRONLY);
+    dup2(tmp_stdout_f, 1);
+    close(tmp_stdout_f);
+  }
+
+  void TearDown()
+  {
+    fflush(stdout);
+    dup2(stdout_f, 1);
+    close(stdout_f);
+  }
+};
+
+//---------------------------------------------------------------------------
+// Test structures
+//---------------------------------------------------------------------------
+
+struct IdleTest : public StateTest {
```

Comment:
Try and populate these structs if possible. Additionally, for better readability in the test files, these structs should be above the suite of tests that use them. If something is specific to a certain test suite and is repeated throughout tests then add it here :)